### PR TITLE
Fix tearDown in test fixtures

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
Moving parent tearDown to the end of tearDown methods. This allows resources to be released in the correct sequence.

### CODE REVIEW

**Review by SebastianGode — DISMISSED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by Polina-Gubina — DISMISSED** [review]

File: otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py [hunk 1]
```diff
@@ -182,70 +114,50 @@ def _delete_instance(self, instance, as_group):
...
 
     def tearDown(self):
-        super(TestInstance, self).tearDown()
         try:
             self._deinitialize_as_group_with_instance()
-        except openstack.exceptions.SDKException as e:
+        except exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py]: using finally is absolutely not required. Moving parent tearDown to the end makes sense, but finally is not required

File: otcextensions/tests/functional/sdk/dms/v1/test_instance.py [hunk 2]
```diff
@@ -90,7 +89,9 @@ def tearDown(self):
         except openstack.exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
-        self._deinitialize_network()
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dms/v1/test_instance.py:92]: same as above, however if we need finally for _deinitialize - no problem

File: otcextensions/tests/functional/sdk/dms/v1/test_message.py [hunk 3]
```diff
@@ -49,14 +49,15 @@ def setUp(self):
...
     def tearDown(self):
-        super(TestMessage, self).tearDown()
         try:
             for queue in self.queues:
                 if queue.id:
                     self.conn.dms.delete_queue(queue)
         except openstack.exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dms/v1/test_message.py]: same here

File: otcextensions/tests/functional/sdk/dms/v1/test_queue.py [hunk 4]
```diff
@@ -33,14 +33,15 @@ def setUp(self):
...
     def tearDown(self):
-        super(TestQueue, self).tearDown()
         try:
             for queue in self.queues:
                 if queue.id:
                     self.conn.dms.delete_queue(queue)
         except exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dms/v1/test_queue.py]: same

File: otcextensions/tests/functional/sdk/dns/v2/test_floatingips.py [hunk 5]
```diff
@@ -33,11 +33,11 @@ def setUp(self):
         )
 
     def tearDown(self):
-        super(TestFloatingIps, self).tearDown()
         self.client.unset_floating_ip(
             floating_ip=self.floatingip,
         )
         self.conn.network.delete_ip(self.floating_ip)
+        super(TestFloatingIps, self).tearDown()
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_floatingips.py:40]: yupp, this way

File: otcextensions/tests/functional/sdk/dns/v2/test_nameservers.py [hunk 6]
```diff
@@ -34,14 +34,15 @@ def setUp(self):
...
     def tearDown(self):
-        super(TestNameservers, self).tearDown()
         try:
             for zone in self.zones:
                 if zone:
                     self.client.delete_zone(zone)
         except openstack.exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_nameservers.py]: and again here unrequired finally

File: otcextensions/tests/functional/sdk/dns/v2/test_recordsets.py [hunk 7]
```diff
@@ -35,14 +35,15 @@ def setUp(self):
...
     def tearDown(self):
-        super(TestRecordsets, self).tearDown()
         try:
             for zone in self.zones:
                 if zone:
                     self.client.delete_zone(zone)
         except openstack.exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_recordsets.py]: same

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py [hunk 8]
```diff
@@ -29,13 +29,14 @@ def setUp(self):
...
 
     def tearDown(self):
-        super(TestZone, self).tearDown()
         try:
             if self.zone:
                 self.client.delete_zone(self.zone)
         except openstack.exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: same

File: otcextensions/tests/functional/sdk/kms/v1/test_data_key.py [hunk 9]
```diff
@@ -38,6 +37,8 @@ def tearDown(self):
         except exceptions.SDKException as e:
             self.warning = _logger.warning('Got exception during '
                                            'clearing resources %s' % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/kms/v1/test_data_key.py]: same

File: otcextensions/tests/functional/sdk/auto_scaling/v1/base.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        super(BaseASTest, self).setUp()
+        self.infra = self.create_test_infra()
+
+    def tearDown(self):
+        try:
+            self.delete_test_infra(self.infra)
+        except exceptions.SDKException as e:
+            _logger.warning('Got exception during clearing resources %s'
+                            % e.message)
+        finally:
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/auto_scaling/v1/base.py]: finally is not necessary here

**Review by gtema — CHANGES_REQUESTED** [review]
Since you anyway need to address some nits - please improve description of the PR to explain why you do this

File: otcextensions/tests/functional/sdk/auto_scaling/v1/base.py (hunk 10 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/auto_scaling/v1/base.py]: Done

File: otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py (hunk 1 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py]: Done

File: otcextensions/tests/functional/sdk/dms/v1/test_instance.py (hunk 2 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dms/v1/test_instance.py:92]: In this case "finally" used mostly for _deinitialize, I think we should leave it as is.

File: otcextensions/tests/functional/sdk/dms/v1/test_message.py (hunk 3 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dms/v1/test_message.py]: Done

File: otcextensions/tests/functional/sdk/dms/v1/test_queue.py (hunk 4 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dms/v1/test_queue.py]: Done

File: otcextensions/tests/functional/sdk/dns/v2/test_floatingips.py (hunk 5 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dns/v2/test_floatingips.py:40]: 👍

File: otcextensions/tests/functional/sdk/dns/v2/test_nameservers.py (hunk 6 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dns/v2/test_nameservers.py]: Done

File: otcextensions/tests/functional/sdk/dns/v2/test_recordsets.py (hunk 7 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dns/v2/test_recordsets.py]: Done

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py (hunk 8 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: Done

File: otcextensions/tests/functional/sdk/kms/v1/test_data_key.py (hunk 9 above)
Comment (irina-pereiaslavskaia) [reply otcextensions/tests/functional/sdk/kms/v1/test_data_key.py]: Done

**Review by Polina-Gubina — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by gtema — APPROVED** [review]

**Pull request merged**